cf/regrid/regrid.py

```diff
@@ -281,17 +345,47 @@ def regrid(
     # ----------------------------------------------------------------
     if isinstance(dst, RegridOperator):
         regrid_operator = dst
-        dst = regrid_operator.dst
+        dst = regrid_operator.dst.copy()
         method = regrid_operator.method
         dst_cyclic = regrid_operator.dst_cyclic
         dst_axes = regrid_operator.dst_axes
+        dst_z = regrid_operator.dst_z
+        src_z = regrid_operator.src_z
+        ln_z = regrid_operator.ln_z
         if regrid_operator.coord_sys == "Cartesian":
             src_axes = regrid_operator.src_axes
 
         create_regrid_operator = False
     else:
         create_regrid_operator = True
 
+        # Parse the z, src_z, dst_z, and ln_z parameters
+        if z is not None:
+            if dst_z is None:
+                dst_z = z
+            else:
+                raise ValueError("Can't set both 'z' and 'dst_z'")
+
+            if src_z is None:
+                src_z = z
+            else:
+                raise ValueError("Can't set both 'z' and 'src_z'")
+
+        elif (src_z is None and dst_z is not None) or (
+            src_z is not None and dst_z is None
+        ):
```

Comment:
Much better, but I'd rather go with `(src_z is None) != (dst_z is None)` to avoid overloading the bitwise operator in a Boolean context:  36b6128b16c4c722c7691f8d13a1b1d0fe654348